src/sentry/incidents/logic.py

```diff
@@ -289,27 +294,26 @@ def build_incident_query_params(incident, start=None, end=None, windowed_stats=F
         params["project_id"] = project_ids
 
     snuba_query = incident.alert_rule.snuba_query
-    entity_subscription = get_entity_subscription_for_dataset(
-        dataset=QueryDatasets(snuba_query.dataset),
-        aggregate=snuba_query.aggregate,
-        time_window=snuba_query.time_window,
-        extra_fields={"org_id": incident.organization.id, "event_types": snuba_query.event_types},
-    )
-    snuba_filter = build_snuba_filter(
-        entity_subscription,
+    snuba_filter = entity_subscription.build_snuba_filter(
         snuba_query.query,
         snuba_query.environment,
         params=params,
     )
+    time_conditions = [
+        [entity_subscription.time_col, ">=", snuba_filter.start],
+        [entity_subscription.time_col, "<", snuba_filter.end],
+    ]
 
     return {
-        "dataset": Dataset(snuba_query.dataset),
-        "start": snuba_filter.start,
-        "end": snuba_filter.end,
-        "conditions": snuba_filter.conditions,
+        "dataset": snuba_query.dataset,
+        "project": project_ids,
+        "project_id": project_ids,
+        "conditions": snuba_filter.conditions + time_conditions,
         "filter_keys": snuba_filter.filter_keys,
         "having": [],
         "aggregations": snuba_filter.aggregations,
+        "limit": 10000,
```

Comment:
Is this limit related to rest of the changes in this PR, or did you add it just for good practice?

Reply:
It was already there, I just moved it https://github.com/getsentry/sentry/pull/30788/files#diff-54ef12aa467e9e032fcdeb5d98079424d01fb28db982938d9e8db849ae78dd5eL373

Reply:
Right, thanks 👍